### ISSUE TITLE
Allow automatic locking for Office files

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/desktop#5893]

**Comment by jospoortvliet** [comment]
I thought this feature was already there? In the announcement we wrote:
"The Desktop Client introduces automated file locking, where it will detect when you locally open a file and locks it on the server to avoid editing conflicts. When you are closing your application, the file is unlocked again. There are also a lot of UI improvements, like the addition of some share settings (hide download, upload only and more) and a number of performance improvements."

or is this something different?

**Comment by jospoortvliet** [comment]
so I think that text is wrong - just checking, I see files get locked when edited online (made read-only locally) but the other way around, editing locally does not lock online. Bit of lsof magic needed I guess 🙈

**Comment by alexanderdd** [comment]
Not sure if this should be discussed here, in the pull request https://github.com/nextcloud/desktop/pull/5986 or in a new issue? @tobiasKaminsky 

I tested this with three other people and it did not work :neutral_face: 

Setup:

- Server: Nextcloud 27.1.2
- Collabora CODE 23.05.5.20231013
- Person A: Linux Mint 20.3 with Appimage client 3.10, Libreoffice 6.4.7.2
- Person B: Windows 10, client 3.10 (virtual files enabled and later disabled – neither worked), MS Office 2309
- Person C: macOS ventura 13.4.1, Nextcloud client 3.10, MS Office 16.78

Scenarios:

- Person A opens .odt locally with Libreoffice, then opens the same file in web – no warning
- Person A opens .odt in web, then opens the same file locally with Libreoffice – no warning
- Person B opens .docx locally with MS Office, Person C opens the same file locally with MS Office – no warning
- Person B opens .docx locally with MS Office, Person C opens the same file in web – no warning
- Person B opens .docx in web, Person C opens the same file locally with MS Office – no warning

Conclusion:

I conclude that in our setup, none of the file locks worked, neiter for Libreoffice nor for MS Office. We saw the lockfiles for MS Office (~$ in the beginning) being synced both in web and locally, but it did not lead to a lock/warning.

In the pull request https://github.com/nextcloud/desktop/pull/5986 people said that there are problems with ubuntu (so probably also with Linux Mint). But the other scenarios should have worked, right?

I can provide more details if necessary.

**Comment by jabo59** [comment]
I have the same problem and the way nextcloud handles this creates a lot of confusion among my users.

**Comment by AT-StephenDetomasi** [comment]
Just wanted to chime in that I'm also seeing inconsistent behaviour here. Anyone tried the latest versions to see if that improves anything?

**Comment by jabo59** [comment]
My problem remains. I‘m using v28.0.3 with app „temporary file locks“ activated.

**Comment by AT-StephenDetomasi** [comment]
Problem appears to be ongoing. It's problematic to the point where we are looking at switching away from NextCloud. Has anyone found a suitable workaround?

**Comment by jabo59** [comment]
No. I‘m currently trying with file locking disabled. I plan also to check this with nextcloud v29. But this will take time.

**Comment by AT-StephenDetomasi** [comment]
NextCloud 29 RC is available now. I don't see any major changes though so I'm not going to hold my breath.

**Comment by thprv** [comment]
Not working on NC 27.1.5 and client 3.13.3 on macOS either.

**Comment by AT-StephenDetomasi** [comment]
Interesting to hear people do have it working. We are in the process of migrating away from Nextcloud over this issue, would love to be able to cancel the project if someone can get this solved.

**Comment by thprv** [comment]
After our hoster made some updates, it is working for me now (after conducting a short experiment, more to follow).

Nextcloud: 28.0.7.5
Nextcloud-Client: 3.13.3
Temporary Files lock: 28.0.6

Will have to check if poll/sync frequency defaults for the client work out in our office.
